core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/impl/StrictEvaluationStrategy.java

```diff
@@ -1432,6 +1423,20 @@ public Value evaluate(LangMatches node, BindingSet bindings)
 
 	}
 
+	public static boolean langTagMatchesRange(String langTag, String langRange) {
+		boolean result = false;
+		if (langRange.equals("*")) {
+			result = langTag.length() > 0;
+		} else if (langTag.length() == langRange.length()) {
+			result = langTag.equalsIgnoreCase(langRange);
+		} else if (langTag.length() > langRange.length()) {
+			// check if the range is a prefix of the tag
+			String prefix = langTag.substring(0, langRange.length());
+			result = prefix.equalsIgnoreCase(langRange) && langTag.charAt(langRange.length()) == '-';
+		}
+		return result;
+	}
+
```

Comment:
Good find that we are reusing this, but I don't think is the right place for this static method. I suggest we move this to the `Literals` utility class (which has other language-tag related utility methods as well, so it's a natural spot).